lib/Controller/RecommendationApiController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Recommendations\Controller;
+
+use OCA\Recommendations\AppInfo\Application;
+use OCA\Recommendations\Service\RecommendationService;
+use OCP\AppFramework\ApiController;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\IConfig;
+use OCP\IRequest;
+use OCP\IUserManager;
+
+class RecommendationApiController extends ApiController {
```

Comment:
I think this should be using OCSController instead: https://docs.nextcloud.com/server/latest/developer_manual/basics/controllers.html#ocs

Reply:
The first sentence in the info box is saying: "This is purely for compatibility reasons. If you are planning to offer an external API, go for a REST APIs instead.".

Reply:
hmm yeah, not completely sure. I remember seeing some new stuff using OCSController. It's probably better to confirm with someone involved in Nextcloud since a longer time :)

@ChristophWurst do you know what is the recommended way?

Reply:
Definitely should be an OCS controller, so all our clients can hook it up and use it in the future

Reply:
So I guess we should fix the docs

Reply:
Who will maintain this API for clients if they start using it? I won't.

Reply:
Well then what is this API needed for?

Reply:
As per https://github.com/nextcloud/recommendations/issues/457 they want to use it in the Outlook plugin, so yeah that is exactly why we have OCS apis and it should be used in this case.

Reply:
I transformed it to an OCS controller, but I don't know who will maintain it in the future :wink:

Reply:
@sualko please document the API at https://docs.nextcloud.com/server/latest/developer_manual/client_apis/OCS/index.html.

We'll allow the API but there are no guarantees for future compatibility, hence please flag as experimental.

Reply:
Also @sualko can you ping us or @ChristophWurst respectively when this is fully integrated and "stable". Reason is we would then merge it when the docs are in but wouldn't publish a new release yet, so in case you need to do some more changes during integration that could still be done and there wouldn't be any scenario that would require migration logic for app upgrades, since this wasn't published yet, which would safe everybody some time.

If this is already finished and works fine from your perspective, please let us know.